### PR TITLE
Adds checking env variable for github actions for 6.x

### PR DESCRIPTION
**Pull request by cerrussell**
### Changes

- Add checking of an environment variable for use with the [AppThreat/dep-scan-action](https://github.com/appthreat/dep-scan-action). User will have to run action with the value "I have sponsored OWASP-dep-scan.".
- Updated workflow actions.

### CODE REVIEW

File: .github/workflows/dockertests.yml [hunk 1]
```diff
@@ -67,10 +67,10 @@ jobs:
         chmod +x contrib/free_disk_space.sh
         ./contrib/free_disk_space.sh
     - name: Set up JDK
-      uses: actions/setup-java@v3
+      uses: actions/setup-java@v4
       with:
         distribution: 'zulu'
-        java-version: '19'
+        java-version: '20'
```
Comment (prabhu) [inline .github/workflows/dockertests.yml:73]: 21

**Review by prabhu — COMMENTED** [review]
This is good. But initially lets use the 5.x for the action

File: depscan/cli.py [hunk 2]
```diff
@@ -744,6 +744,16 @@ def main():
         pkg_vulnerabilities,
         pkg_group_rows,
     ) = (None, None, None, None, None, None)
+    if os.getenv("GITHUB_ACTIONS").lower() == "__appthreat_dep-scan-action" \
+        and not os.getenv("INPUT_THANK_YOU") == ("I have sponsored "
+                                                 "OWASP-dep-scan."):
+        LOG.error("You have not indicated that you have sponsored "
```
Comment (prabhu) [inline depscan/cli.py:750]: May be info within a rich panel?

**Comment by cerrussell** [comment]
> This is good. But initially lets use the 5.x for the action

I didn't see this and already squashed the PR that would change this in dep-scan-action. However, I changed now - hopefully no issues as that is not what I tested.

File: depscan/cli.py (hunk 2 above)
Comment (cerrussell) [reply depscan/cli.py:750]: @prabhu Is this ok? Wasn't sure about title.

image

**Review by prabhu — APPROVED** [review]

**Pull request merged**